rero_ils/modules/items/utils.py

```diff
@@ -106,7 +106,13 @@ def get_provisional_items_pids_candidate_to_delete():
         .filter('range', total_amount={'gt': 0})\
         .source('item')
     # list of item pids with open fees
-    item_pids_with_fees = [hit.item.pid for hit in query_fees.scan()]
+    item_pids_with_fees = []
+    try:
+        for hit in query_fees.scan():
+            if hit.item:
+                item_pids_with_fees.append(hit.item.pid)
+    except AttributeError:
+        pass
```

Comment:
It seems `try...except` is badly placed. If you have a problem reading the second hit of the `scan()`, no other hit will be checked.
Don't you think it's better to just add an `exists` filter on ES query : 
```python
query_fees = PatronTransactionsSearch()\
        .filter('term', status='open')\
        .filter('exists', field='item')\
        .filter('range', total_amount={'gt': 0})\
        .source('item')
item_pids_with_fees = [hit.item.pid for hit in query_fees.scan()]
```